### PR TITLE
[BUG](date_trunc) fix date_trunc function only handle lower string

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes
date_trunc function only handle lower string, so need convert this param to lower case.

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

File: be/src/vec/functions/function_timestamp.cpp [hunk 1]
```diff
@@ -423,10 +423,12 @@ struct DateTrunc {
                                          const StringRef& rdata, PaddedPODArray<ArgType>& res,
                                          NullMap& null_map, size_t input_rows_count) {
         res.resize(input_rows_count);
+        std::string lower_str(rdata.data, rdata.size);
```
Comment (yiguolei) [inline be/src/vec/functions/function_timestamp.cpp:426]: add regression test please

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 44.77 seconds
 stream load tsv:          512 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17162084832 Bytes

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.31 seconds
 stream load tsv:          514 seconds loaded 74807831229 Bytes, about 138 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          30.1 seconds inserted 10000000 Rows, about 332K ops/s
 storage size: 17162270464 Bytes

**Comment by zhangstar333** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 48.37 seconds
 stream load tsv:          514 seconds loaded 74807831229 Bytes, about 138 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          66 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          33 seconds loaded 861443392 Bytes, about 24 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17162345718 Bytes

**Review by yiguolei — APPROVED** [review]
lgtm

File: regression-test/suites/query_p0/sql_functions/datetime_functions/test_date_trunc.groovy [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    qt_select_date_trunc_hour """ SELECT k11, date_trunc(k11,'Hour') FROM baseall order by k1,k2,k3;"""
+    qt_select_date_trunc_day """ SELECT k11, date_trunc(k11,'DAY') FROM baseall order by k1,k2,k3;"""
+    qt_select_date_trunc_week """ SELECT k11, date_trunc(k11,'Week') FROM baseall order by k1,k2,k3;"""
+    qt_select_date_trunc_month """ SELECT k11, date_trunc(k11,'MONTH') FROM baseall order by k1,k2,k3;"""
+    qt_select_date_trunc_quarter """ SELECT k11, date_trunc(k11,'quarter') FROM baseall order by k1,k2,k3;"""
+    qt_select_date_trunc_year """ SELECT k11, date_trunc(k11,'YeaR') FROM baseall order by k1,k2,k3;"""
+
+    try {
+        sql """ SELECT date_trunc(k11,k7) FROM baseall ; """
+    } catch (Exception e) {
```
Comment (BiteTheDDDDt) [inline regression-test/suites/query_p0/sql_functions/datetime_functions/test_date_trunc.groovy:65]: If no exception is thrown?

File: regression-test/data/query_p0/sql_functions/datetime_functions/test_date_trunc.out [hunk 3]
```diff
@@ -0,0 +1,145 @@
+-- This file is automatically generated. You should know what you did if you want to edit this
+-- !select_date_trunc_second --
+\N	\N
+1989-03-21T13:00	1989-03-21T13:00
+1989-03-21T13:00	1989-03-21T13:00
+2000-01-01T00:00	2000-01-01T00:00
+2015-03-13T10:30	2015-03-13T10:30
+2015-03-13T12:36:38	2015-03-13T12:36:38
```
Comment (BiteTheDDDDt) [inline regression-test/data/query_p0/sql_functions/datetime_functions/test_date_trunc.out:8]: why the result looks unordered？

**Review by morrySnow — APPROVED** [review]

**Pull request merged**